### ISSUE TITLE
--multi-user still breaks if multiple users prompt at the same time

### ISSUE THREAD

**Issue by Fusseldieb** [oobabooga/text-generation-webui#3490]
### Describe the bug

Basically the title. If I and a friend, at the same time, ask something from the WebUI (on different computers), it starts generating until it receives the second request and then it simply stops for both. It just _stops_ generating.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

1. Enable --multi-user on "Session" and restart
2. Open WebUI on my PC
3. Open WebUI on friends PC
4. We both type something it and press enter at the same time
5. It starts generating and freezes after a couple of words.

### Screenshot

![image](https://github.com/oobabooga/text-generation-webui/assets/4715129/e4f8ec24-e306-47a3-b314-7bddae6bad30)

### Logs

```shell
Traceback (most recent call last):
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\text-generation-webui\modules\text_generation.py", line 327, in generate_reply_custom
    for reply in shared.model.generate_with_streaming(question, state):
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\text-generation-webui\modules\exllama.py", line 102, in generate_with_streaming
    token = self.generator.gen_single_token()
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\installer_files\env\lib\site-packages\exllama\generator.py", line 353, in gen_single_token
    self.apply_rep_penalty(logits)
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\installer_files\env\lib\site-packages\exllama\generator.py", line 335, in apply_rep_penalty
    cuda_ext.ext_apply_rep_penalty_mask_cpu(self.sequence,
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\installer_files\env\lib\site-packages\exllama\cuda_ext.py", line 110, in ext_apply_rep_penalty_mask_cpu
    apply_rep_penalty(sequence, penalty_max, sustain, decay, logits)
TypeError: apply_rep_penalty(): incompatible function arguments. The following argument types are supported:
    1. (arg0: torch.Tensor, arg1: float, arg2: int, arg3: int, arg4: torch.Tensor) -> None

Invoked with: tensor([], size=(1, 0), dtype=torch.int64), 1.17, -1, 128, None
Output generated in 0.01 seconds (0.00 tokens/s, 0 tokens, context 72, seed 509148073)
Traceback (most recent call last):
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\text-generation-webui\modules\text_generation.py", line 327, in generate_reply_custom
    for reply in shared.model.generate_with_streaming(question, state):
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\text-generation-webui\modules\exllama.py", line 102, in generate_with_streaming
    token = self.generator.gen_single_token()
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\installer_files\env\lib\site-packages\exllama\generator.py", line 353, in gen_single_token
    self.apply_rep_penalty(logits)
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\installer_files\env\lib\site-packages\exllama\generator.py", line 335, in apply_rep_penalty
    cuda_ext.ext_apply_rep_penalty_mask_cpu(self.sequence,
  File "C:\Users\vstil\Downloads\oobabooga_windows\oobabooga_windows\installer_files\env\lib\site-packages\exllama\cuda_ext.py", line 110, in ext_apply_rep_penalty_mask_cpu
    apply_rep_penalty(sequence, penalty_max, sustain, decay, logits)
TypeError: apply_rep_penalty(): incompatible function arguments. The following argument types are supported:
    1. (arg0: torch.Tensor, arg1: float, arg2: int, arg3: int, arg4: torch.Tensor) -> None

Invoked with: tensor([], size=(1, 0), dtype=torch.int64), 1.17, -1, 128, None
Output generated in 0.00 seconds (0.00 tokens/s, 0 tokens, context 71, seed 531912770)
```

### System Info

```shell
Windows 11 x64
NVIDIA GeForce 2080 8GB VRAM (Notebook)
8GB RAM
```

**Comment by oobabooga** [comment]
Can you check if this fixes it? https://github.com/oobabooga/text-generation-webui/commit/6d354bb50b62d6ded5431a9097e6ca0db748678d

**Comment by Fusseldieb** [comment]
> Can you check if this fixes it? [6d354bb](https://github.com/oobabooga/text-generation-webui/commit/6d354bb50b62d6ded5431a9097e6ca0db748678d)

Thanks! I've just updated it and now it _kinda_ works. It doesn't error out anymore, at least. However, it generates only

**Comment by KartavyaBagga** [comment]
Did you find any solution to calling asynchronously ?
